kedro-datasets/kedro_datasets_experimental/opik/opik_prompt_dataset.py

```diff
@@ -0,0 +1,596 @@
+"""Opik prompt dataset for seamless integration between local prompt files (JSON/YAML) and Opik prompt management"""
+
+import hashlib
+import json
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Literal, Union
+
+from kedro.io import AbstractDataset, DatasetError
+
+if TYPE_CHECKING:
+    from langchain_core.prompts import ChatPromptTemplate
+
+    from kedro_datasets.json import JSONDataset
+    from kedro_datasets.yaml import YAMLDataset
+
+from opik import Opik, Prompt
+
+from kedro_datasets._typing import JSONPreview
+
+# Supported file extensions for prompt storage
+SUPPORTED_FILE_EXTENSIONS = {".json", ".yaml", ".yml"}
+
+# Valid parameter values
+VALID_PROMPT_TYPES = {"chat", "text"}
+VALID_SYNC_POLICIES = {"local", "remote", "strict"}
+VALID_MODES = {"langchain", "sdk"}
+
+# Logger for this module
+logger = logging.getLogger(__name__)
+
+
+def _hash(data: str | list) -> str:
+    """Return SHA-256 hash of a prompt (string or list of messages)."""
+    return hashlib.sha256(json.dumps(data, sort_keys=True).encode()).hexdigest()
+
+
+def _get_content(data: str | list) -> str:
+    """
+    Extract comparable text content from a prompt.
+
+    - If string: return as-is.
+    - If list of messages: join their `content`.
+    """
+    if isinstance(data, str):
+        return data
+    return "\n".join(msg["content"] for msg in data)
+
+
+class OpikPromptDataset(AbstractDataset):
+    """Kedro dataset for managing prompts with Opik versioning and synchronisation.
+
+    This dataset provides seamless integration between local prompt files (JSON/YAML)
+    and Opik's prompt management system, supporting version control, metadata tracking,
+    and different synchronisation policies.
+
+    On save: Creates a new version of prompt in Opik with the local data.
+    On load: Synchronises based on sync_policy and returns raw Opik object
+    (sdk mode) or LangChain ChatPromptTemplate (langchain mode).
+
+    Sync policies:
+        - local: Local file takes precedence (default). Load_args are ignored with warning
+          since local files are the source of truth.
+        - remote: Opik version takes precedence. Load_args are respected if supported.
+        - strict: Error if local and remote differ. Load_args are respected if supported.
+
+    Examples:
+        Using catalog YAML configuration:
+
+        .. code-block:: yaml
+
+            # Local sync policy - local files are source of truth
+            customer_prompt:
+              type: opik.OpikPromptDataset
```

Comment:
```suggestion
              type: kedro_datasets_experimental.opik.OpikPromptDataset
```
otherwise, it fails to find it